client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchDetails.tsx

```diff
@@ -0,0 +1,600 @@
+    setIsEditing(true)
+
+    // Open the first editable sheet if not already on an editable sheet
+    if (sheetFields.length > 1 && selectedTabId === VOTE_TOTALS_TAB_ID) {
+      selectTabWithSheetIndex(0)
+    }
+  }
+
+  const addSheet = () => {
+    let newSheetName = defaultResultTallySheetName(sheetFields.length)
+    function doesNewSheetNameConflict() {
+      return sheetFields.some(
+        (_, i) => getValues(`resultTallySheets[${i}].name`) === newSheetName
+      )
+    }
+
+    let incrementToAvoidConflict = 0
+    while (doesNewSheetNameConflict()) {
+      incrementToAvoidConflict += 1
+      newSheetName = defaultResultTallySheetName(
+        sheetFields.length + incrementToAvoidConflict
+      )
+    }
+
+    sheetFieldArrayMethods.append({
+      name: newSheetName,
+      results: {},
+    })
+    // We want to auto-select the new sheet but don't yet have the react-hook-form useFieldArray ID
+    // to select it, so we do so via a useEffect hook
+    setAutoSelectLastTabOnceNumSheetsIs(sheetFields.length + 1)
+  }
+
+  const openAdditionalSheetActions = () => {
+    setAreAdditionalSheetActionsOpen(true)
+  }
+
+  const closeAdditionalSheetActions = () => {
+    setAreAdditionalSheetActionsOpen(false)
+  }
+
+  const initiateSheetRename = () => {
+    setDraftSheetName(getValues(`resultTallySheets[${currentSheetIndex}].name`))
+    setIsRenamingSheet(true)
+  }
+
+  const confirmSheetRename = () => {
+    setValue(`resultTallySheets[${currentSheetIndex}].name`, draftSheetName, {
+      shouldDirty: true,
+    })
+    setIsRenamingSheet(false)
+    setAreAdditionalSheetActionsOpen(false)
+    setDraftSheetName('')
+  }
+
+  const cancelSheetRename = () => {
+    setIsRenamingSheet(false)
+    setAreAdditionalSheetActionsOpen(false)
+    setDraftSheetName('')
+  }
+
+  const removeSheet = () => {
+    const numSheetsAfterDeletion = sheetFields.length - 1
+
+    if (numSheetsAfterDeletion === 1 || currentSheetIndex === 0) {
+      selectVoteTotalsTab()
+    } else {
+      // Open the sheet before the to-be-deleted sheet
+      selectTabWithSheetIndex(Math.max(currentSheetIndex - 1, 0))
+    }
+
+    // If we drop back down to 1 sheet, rename that sheet "Sheet 1" under the hood
+    if (numSheetsAfterDeletion === 1) {
+      const remainingIndex = currentSheetIndex === 1 ? 0 : 1
+      setValue(`resultTallySheets[${remainingIndex}].name`, 'Sheet 1')
+    }
+
+    sheetFieldArrayMethods.remove(currentSheetIndex)
+    setAreAdditionalSheetActionsOpen(false)
+  }
+
+  const onValidSubmit: SubmitHandler<IBatchRoundDataEntryFormState> = async ({
+    resultTallySheets,
+  }) => {
+    if (!batch) {
+      return
+    }
+    try {
+      await saveBatchResults(resultTallySheets)
+    } catch {
+      // Errors are automatically toasted by the queryClient
+      setIsEditing(true)
+      return
+    }
+    selectVoteTotalsTab()
+    // Reset the form's isDirty value back to false
+    reset({ resultTallySheets })
```

Comment:
react-hook-form (a bit unintuitively) doesn't reset `isDirty` to false after successful submission